mantis-runtime/src/main/java/io/mantisrx/runtime/parameter/SourceJobParameters.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.runtime.parameter;
+
+import com.mantisrx.common.utils.MantisSSEConstants;
+import io.mantisrx.runtime.Context;
+import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import io.mantisrx.shaded.com.fasterxml.jackson.annotation.JsonProperty;
+import io.mantisrx.shaded.com.fasterxml.jackson.core.type.TypeReference;
+import io.mantisrx.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class SourceJobParameters {
+    public static final String MANTIS_SOURCEJOB_TARGET_KEY = "target";
```

Comment:
This assumes a multi source job connector. For single source job connectors, we would look at the sourceJobName job param.
For MRE v2 app stream connector, there is no source job name in the job params.

We should probably make this more generic for 
1) Source job name config: either identify the source job names that need to be queried (using another job param value or app stream etc) or provide the source job names in the autoscaling config 
2) the metric group patterns to match against

Reply:
(1) makes sense. The various connectors may use different params for both `sourceJobName` and `clientId`. There is no clean way to support all of them. I will add a system param called `mantis.jobmaster.autoscale.sourcejob.target` that can be used as override.

(2) what's the use case? Are there other job-to-job transport outside of `ServerSentEventRequestHandler` that can emit drops?

Reply:
2) no current use case, this is more forward looking if we add a websocket/grpc sink in future.